src/main/java/edu/project/yourpackage/WeatherApi.java

```diff
@@ -0,0 +1,47 @@
+//API key: 23c541de6a2105ebf03f39791a6b7cd1
+package edu.project.yourpackage;
+
+import com.google.gson.Gson;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class WeatherApi {
+    private static final String API_KEY = "23c541de6a2105ebf03f39791a6b7cd1";
+    private static final String CURRENT_WEATHER_URL = "http://api.openweathermap.org/data/2.5/weather";
+    private static final String FORECAST_WEATHER_URL = "http://api.openweathermap.org/data/2.5/forecast";
+
+    public WeatherCurrentData getCurrentWeather(String city) throws IOException {
+        String urlString = CURRENT_WEATHER_URL + "?q=" + city + "&appid=" + API_KEY;
+        return fetchWeatherData(urlString, WeatherCurrentData.class);
+        
+    }
+
+    public WeatherForecast5Data getWeatherForecast(String city) throws IOException{
+        String urlString = FORECAST_WEATHER_URL + "?q=" + city + "&appid=" + API_KEY;
+        return fetchWeatherData(urlString, WeatherForecast5Data.class);
+       
+    }
```

Comment:
aside from being able to accept a `String city` parameter, the methods should also be able to accept a `double latitude, double longitude` parameter (use method overloading).

alternatively, we can use the `GeoPosition` class instead of having two seperate `double` parameters for the latitude & longitude.